tests/5.0/target/test_target_device.c

```diff
@@ -0,0 +1,81 @@
+//===--- test_target_device.c ----------------------------------------------===//
+//
+// OpenMP API Version 5.0 Nov 2018
+//
+// This test checks the target construct with device clause where device-
+// modifier is either ancestor or device_num. If no device_modifier is 
+// present, the behavior is the same as if device_num were present.
+//
+////===---------------------------------------------------------------------===//
+
+#include <stdlib.h>
+#include <stdio.h>
+#include <omp.h>
+#include "ompvv.h"
+
+#define N 1028
+
+int test_target_device_ancestor() {
+
+    int i, which_device;
+    int a[N];
+    int errors = 0; 
+
+    for (int i = 0; i < N; i++) {
+        a[i] = i;
+    }
+
+    #pragma omp target device(ancestor: 1) map(tofrom: a, which_device) 
+    {
+        for (int i = 0; i < N; i++) {
+            a[i] = a[i] + 2;
+        }
+ 
+        which_device = omp_is_initial_device();
+    }
+
+    OMPVV_TEST_AND_SET_VERBOSE(errors, which_device != 1);
+    OMPVV_WARNING_IF(which_device != 1, "Target region was executed on device. Due to ancestor device-modifier,"
+                                         "this region should execute on host");
+
+    return errors;
+
+}
+
+int test_target_device_device_num() {
+    
+    int i, which_device;
+    int b[N];
+    int errors = 0; 
+
+    for (int i = 0; i < N; i++) {
+        b[i] = i;
+    }
+
+    #pragma omp target device(device_num: 1) map(tofrom: b, which_device) 
```

Comment:
In all practical situations the assumption that device_num == 1 will work, although device numbers are implementation defined. Ideally the device_num should be set to a value between 0 - omp_get_num_devices() and the value should not be equal to omp_get_device_num() on host.